services/src/pro/users/hashmap_userdb.rs

```diff
@@ -102,6 +105,50 @@ impl UserDb for HashMapUserDb {
         }
     }
 
+    async fn login_external(&self, user: ExternalUserClaims, duration: Duration) -> Result<UserSession> {
```

Comment:
This function is only called after a successful call to resolve_request in the OIDC database. So a user was already authenticated and has a valid external id, as well as email+name+duration associated with it. In practice, this call can only succeed. It is still a necessary call though to connect the external id to the internal user representation in Geo Engine so that projects etc. work.

Reply:
On second thought, it is probably weird if the name/email changes at the ID Provider and the Geo Engine has outdated information. That would require an update here, but that might still be a strict success case, if we define it that way.

Reply:
Okay, so we can leave it as is, right?

Reply:
Yes. I'd say we can rethink this if we work with more claims or actually do something with the email/name.